aea/protocols/dialogue/base.py

```diff
@@ -1136,53 +1202,88 @@ def _get_collection_name(self) -> Optional[str]:
             ]
         )
 
-    def _get_collection(self) -> Optional[SyncCollection]:
+    def _get_collection_instance(self, col_name: str) -> Optional[SyncCollection]:
         """Get sync collection if generic storage available."""
         if (
             not self._skill_component or not self._skill_component.context.storage
         ):  # pragma: nocover
             return None
-        col_name = self._get_collection_name()
-        if not col_name:  # pragma: nocover
-            return None
         return self._skill_component.context.storage.get_sync_collection(col_name)
 
+    def _get_collections(
+        self,
+    ) -> Tuple[Optional[SyncCollection], Optional[SyncCollection]]:
+        """Get active state dialogue collection and terminal collection."""
+        col_name = self._get_collection_name()
+        if not col_name:
+            return (None, None)  # pragma: nocover
+        termina_col_name = f"{col_name}_terminal"
+
+        collection = self._get_collection_instance(col_name)
+        terminal_collection = self._get_collection_instance(termina_col_name)
+        if not collection or not terminal_collection:
```

Comment:
Not sure why we return like this if either of them is missing...

Reply:
discussed